src/main/java/com/exadel/discountwebapp/location/repository/LocationRepository.java

```diff
@@ -1,15 +1,28 @@
 package com.exadel.discountwebapp.location.repository;
 
 import com.exadel.discountwebapp.location.entity.Location;
+import org.springframework.data.jpa.repository.Query;
 import org.springframework.data.repository.CrudRepository;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 import java.util.List;
 
 @Repository
 public interface LocationRepository extends CrudRepository<Location, Long> {
 
-    List<Location> findAllByCountry(String country);
+    @Query(value =
+            "select location.loc_id, " +
+                    "location.country_code, " +
+                    "location.loc_city, " +
+                    "location.loc_address_line," +
+                    "location.loc_created, " +
+                    "location.loc_modified " +
+                    "from location " +
+                    "left join country " +
+                    "on location.country_code = country.country_code " +
+                    "where country.country_code = :countryCode", nativeQuery = true)
+    List<Location> findAllByCountryCode(@Param("countryCode") String countryCode);
```

Comment:
can be replaced with 

List<Location> findAllByCountryCode_CountryCode(String countryCode);

Reply:
"_" is used to separate entities. like "." in java

Reply:
I have tried, previously. Somehow does't work. Throws exception that CountryCode shound be a Country class. If use JPQL with class entities - the same result.

Reply:
Error like this:

Traversed path: Location.country.; nested exception is java.lang.IllegalArgumentException: Failed to create query for method public abstract java.util.List com.exadel.discountwebapp.location.repository.LocationRepository.findAllByCountryCode_CountryCode(java.lang.String)! No property code found for type Country! Traversed path: Location.country.